Libgit2 ERROR: AssertionError: name == "helper"

When cloning a package (upon adding a package from Github using a url), I have found this error:
](https://discourse.julialang.org/t/libgit2-error-assertionerror-name-helper/82783)

I am still getting this error (Julia 1.7.2 on Windows 10):
```

(SteelSphere) pkg> add https://github.com/PetrKryslUCSD/BEMScat.jl.git
     Cloning git-repo `https://github.com/PetrKryslUCSD/BEMScat.jl.git`
ERROR: AssertionError: name == "helper"
Stacktrace:
  [1] credential_helpers(cfg::LibGit2.GitConfig, cred::LibGit2.GitCredential)
    @ LibGit2 C:\Users\pkonl\VSCode_Julia_portable\assets\julia-1.7.2\share\julia\stdlib\v1.7\LibGit2\src\gitcredential.jl:224
  [2] fill!(cfg::LibGit2.GitConfig, cred::LibGit2.GitCredential)
    @ LibGit2 C:\Users\pkonl\VSCode_Julia_portable\assets\julia-1.7.2\share\julia\stdlib\v1.7\LibGit2\src\gitcredential.jl:148
  [3] GitCredential
    @ C:\Users\pkonl\VSCode_Julia_portable\assets\julia-1.7.2\share\julia\stdlib\v1.7\LibGit2\src\gitcredential.jl:30 [inlined]
  [4] authenticate_userpass(libgit2credptr::Ptr{Ptr{Nothing}}, p::LibGit2.CredentialPayload)
    @ LibGit2 C:\Users\pkonl\VSCode_Julia_portable\assets\julia-1.7.2\share\julia\stdlib\v1.7\LibGit2\src\callbacks.jl:193
  [5] credentials_callback(libgit2credptr::Ptr{Ptr{Nothing}}, url_ptr::Cstring, username_ptr::Cstring, allowed_types::UInt32, p::LibGit2.CredentialPayload)
    @ LibGit2 C:\Users\pkonl\VSCode_Julia_portable\assets\julia-1.7.2\share\julia\stdlib\v1.7\LibGit2\src\callbacks.jl:328
  [6] credentials_callback(libgit2credptr::Ptr{Ptr{Nothing}}, url_ptr::Cstring, username_ptr::Cstring, allowed_types::UInt32, payloads::Dict{Symbol, Any})
    @ LibGit2 C:\Users\pkonl\VSCode_Julia_portable\assets\julia-1.7.2\share\julia\stdlib\v1.7\LibGit2\src\callbacks.jl:351
  [7] macro expansion
    @ C:\Users\pkonl\VSCode_Julia_portable\assets\julia-1.7.2\share\julia\stdlib\v1.7\LibGit2\src\error.jl:108 [inlined]
  [8] clone(repo_url::SubString{String}, repo_path::String, clone_opts::LibGit2.CloneOptions)
    @ LibGit2 C:\Users\pkonl\VSCode_Julia_portable\assets\julia-1.7.2\share\julia\stdlib\v1.7\LibGit2\src\repository.jl:459
  [9] clone(repo_url::SubString{String}, repo_path::String; branch::String, isbare::Bool, remote_cb::Ptr{Nothing}, credentials::LibGit2.CachedCredentials, callbacks::Dict{Symbol, Tuple{Ptr{Nothing}, Any}})
    @ LibGit2 C:\Users\pkonl\VSCode_Julia_portable\assets\julia-1.7.2\share\julia\stdlib\v1.7\LibGit2\src\LibGit2.jl:583
 [10] clone(io::Base.TTY, url::String, source_path::String; header::Nothing, credentials::Nothing, kwargs::Base.Pairs{Symbol, Bool, Tuple{Symbol}, NamedTuple{(:isbare,), Tuple{Bool}}})
    @ Pkg.GitTools C:\Users\pkonl\VSCode_Julia_portable\assets\julia-1.7.2\share\julia\stdlib\v1.7\Pkg\src\GitTools.jl:115
 [11] #ensure_clone#2
    @ C:\Users\pkonl\VSCode_Julia_portable\assets\julia-1.7.2\share\julia\stdlib\v1.7\Pkg\src\GitTools.jl:74 [inlined]
 [12] handle_repo_add!(ctx::Pkg.Types.Context, pkg::Pkg.Types.PackageSpec)
    @ Pkg.Types C:\Users\pkonl\VSCode_Julia_portable\assets\julia-1.7.2\share\julia\stdlib\v1.7\Pkg\src\Types.jl:676
 [13] handle_repos_add!(ctx::Pkg.Types.Context, pkgs::Vector{Pkg.Types.PackageSpec})
    @ Pkg.Types C:\Users\pkonl\VSCode_Julia_portable\assets\julia-1.7.2\share\julia\stdlib\v1.7\Pkg\src\Types.jl:743
 [14] add(ctx::Pkg.Types.Context, pkgs::Vector{Pkg.Types.PackageSpec}; preserve::Pkg.Types.PreserveLevel, platform::Base.BinaryPlatforms.Platform, kwargs::Base.Pairs{Symbol, Base.TTY, Tuple{Symbol}, NamedTuple{(:io,), Tuple{Base.TTY}}})
    @ Pkg.API C:\Users\pkonl\VSCode_Julia_portable\assets\julia-1.7.2\share\julia\stdlib\v1.7\Pkg\src\API.jl:248
 [15] add(pkgs::Vector{Pkg.Types.PackageSpec}; io::Base.TTY, kwargs::Base.Pairs{Symbol, Union{}, Tuple{}, NamedTuple{(), Tuple{}}})
    @ Pkg.API C:\Users\pkonl\VSCode_Julia_portable\assets\julia-1.7.2\share\julia\stdlib\v1.7\Pkg\src\API.jl:149
 [16] add(pkgs::Vector{Pkg.Types.PackageSpec})
    @ Pkg.API C:\Users\pkonl\VSCode_Julia_portable\assets\julia-1.7.2\share\julia\stdlib\v1.7\Pkg\src\API.jl:144
 [17] do_cmd!(command::Pkg.REPLMode.Command, repl::REPL.LineEditREPL)
    @ Pkg.REPLMode C:\Users\pkonl\VSCode_Julia_portable\assets\julia-1.7.2\share\julia\stdlib\v1.7\Pkg\src\REPLMode\REPLMode.jl:407
 [18] do_cmd(repl::REPL.LineEditREPL, input::String; do_rethrow::Bool)
    @ Pkg.REPLMode C:\Users\pkonl\VSCode_Julia_portable\assets\julia-1.7.2\share\julia\stdlib\v1.7\Pkg\src\REPLMode\REPLMode.jl:385
 [19] do_cmd
    @ C:\Users\pkonl\VSCode_Julia_portable\assets\julia-1.7.2\share\julia\stdlib\v1.7\Pkg\src\REPLMode\REPLMode.jl:376 [inlined]
 [20] (::Pkg.REPLMode.var"#24#27"{REPL.LineEditREPL, REPL.LineEdit.Prompt})(s::REPL.LineEdit.MIState, buf::IOBuffer, ok::Bool)
    @ Pkg.REPLMode C:\Users\pkonl\VSCode_Julia_portable\assets\julia-1.7.2\share\julia\stdlib\v1.7\Pkg\src\REPLMode\REPLMode.jl:549
 [21] #invokelatest#2
    @ .\essentials.jl:716 [inlined]
 [22] invokelatest
    @ .\essentials.jl:714 [inlined]
 [23] run_interface(terminal::REPL.Terminals.TextTerminal, m::REPL.LineEdit.ModalInterface, s::REPL.LineEdit.MIState)
    @ REPL.LineEdit C:\Users\pkonl\VSCode_Julia_portable\assets\julia-1.7.2\share\julia\stdlib\v1.7\REPL\src\LineEdit.jl:2493
 [24] run_frontend(repl::REPL.LineEditREPL, backend::REPL.REPLBackendRef)
    @ REPL C:\Users\pkonl\VSCode_Julia_portable\assets\julia-1.7.2\share\julia\stdlib\v1.7\REPL\src\REPL.jl:1232
 [25] (::REPL.var"#49#54"{REPL.LineEditREPL, REPL.REPLBackendRef})()
    @ REPL .\task.jl:423

(SteelSphere) pkg> 
```

as is 1.9, it seems
```
Pkg.add(url=s"https://github.com/...")
ERROR: AssertionError: name == "helper"
Stacktrace:
  [1] credential_helpers(cfg::LibGit2.GitConfig, cred::LibGit2.GitCredential)
    @ LibGit2 C:\Users\MrJSa\AppData\Local\Programs\Julia-1.9\share\julia\stdlib\v1.9\LibGit2\src\gitcredential.jl:224
```

--- a/stdlib/LibGit2/src/gitcredential.jl
+++ b/stdlib/LibGit2/src/gitcredential.jl
@@ -219,7 +219,7 @@ function credential_helpers(cfg::GitConfig, cred::GitCredential)
     helpers = GitCredentialHelper[]
 
     # https://git-scm.com/docs/gitcredentials#gitcredentials-helper
-    for entry in GitConfigIter(cfg, r"credential.*\.helper")
+    for entry in GitConfigIter(cfg, r"credential.*\.helper$")
         section, url, name, value = split_cfg_entry(entry)
         @assert name == "helper"
 

--- a/stdlib/LibGit2/test/libgit2-tests.jl
+++ b/stdlib/LibGit2/test/libgit2-tests.jl
@@ -1994,7 +1994,7 @@ mktempdir() do dir
             @test parse(GitCredentialHelper, "store") == GitCredentialHelper(`git credential-store`)
         end
 
-        @testset "empty helper" begin
+        @testset "credential_helpers" begin
             config_path = joinpath(dir, config_file)
 
             # Note: LibGit2.set! doesn't allow us to set duplicates or ordering
@@ -2007,16 +2007,14 @@ mktempdir() do dir
                     [credential]
                         helper = !echo second
                     """)
+                # Git for Windows uses this config (see issue #45693)
+                write(fp,"""
+                    [credential "helperselector"]
+                        selected = manager-core
+                    """)
             end
 
             LibGit2.with(LibGit2.GitConfig(config_path, LibGit2.Consts.CONFIG_LEVEL_APP)) do cfg
-                iter = LibGit2.GitConfigIter(cfg, r"credential.*\.helper")
-                @test LibGit2.split_cfg_entry.(iter) == [
-                    ("credential", "", "helper", "!echo first"),
-                    ("credential", "https://mygithost", "helper", ""),
-                    ("credential", "", "helper", "!echo second"),
-                ]
-
                 expected = [
                     GitCredentialHelper(`echo first`),
                     GitCredentialHelper(`echo second`),

